epsagon/events/pymongo.py

```diff
@@ -33,30 +37,30 @@ def __init__(self, wrapped, instance, args, kwargs, start_time, response,
         :param response: response data
         :param exception: Exception (if happened)
         """
-
         super(PyMongoEvent, self).__init__(start_time)
 
-        documents = args[0]
+        self.resource['operation'] = getattr(wrapped, '__name__')
 
-        self.event_id = 'mongo-{}'.format(str(uuid4()))
-        self.resource['name'] = instance.full_name
+        if not args:
+            input_args = list()
+
+        else:
+            input_args = args[0]
```

Comment:
I think the [0] logic should be done inside the handle_request_payload (according to the operation , you can know what to expect in args)
So basically, you would just send "args" as is to handle_request_payload. The handle_request_payload obviously should be called only if there're any args